x/feemarket/types/resolver.go

```diff
@@ -0,0 +1,39 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// DenomResolver is an interface to convert a given token to the feemarket's base token.
+type DenomResolver interface {
+	// ConvertToBaseToken converts feeCoin into the equivalent amount of the token denominated in baseDenom.
+	ConvertToBaseToken(ctx sdk.Context, feeCoin sdk.Coin, baseDenom string) (sdk.Coin, error)
+}
```

Comment:
curious: Do we want this to accept multiple fee coins? would be interesting to be able to supply a basket of coins that resolve to the base fee.

Reply:
This depends on if we are going to go allow only one denom to be provided.  If we only allow one, then I think this interface makes sense.  otherwise, we should move it to take in Coins